### ISSUE TITLE
Wine (wineboot) hangs when creating the initial prefix (return code 13568)

### ISSUE THREAD

**Issue by WildPenquin** [lutris/lutris#3973]
### Bug description

Most (all?) game installations fail when creating the initial wine prefix. Wine dependencies have been installed and I can rune standalone wine without problems (Arch Linux). Prefixes I create manually do not hang at the same step (very first wine command).

After waiting a very long time, wineboot is killed (?) by lutris.

I get this at least when installing Terraformin Mars, System Shock EE from gog.com and Battle.net from Blizzard, but it seems I can reproduce this with any (of course I certainly didn't try all!) software via Lutris.

I believe wine hanging at this stage means Lutris has set up something wrong in the wine environment, as this is a very basic step and wine prefix creation should not fail. Hence checking "I'm not asking for support with a game or the wine runner.", however I can understand this is a corner case (?). I'm not sure what a user can do if Wine does not work via Lutris but does work otherwise on the same system.

```
$ LANG=C sudo pacman -S --needed wine-staging giflib lib32-giflib libpng lib32-libpng libldap lib32-libldap gnutls lib32-gnutls mpg123 lib32-mpg123 openal lib32-openal v4l-utils lib32-v4l-utils libpulse lib32-libpulse libgpg-error lib32-libgpg-error alsa-plugins lib32-alsa-plugins alsa-lib lib32-alsa-lib libjpeg-turbo lib32-libjpeg-turbo sqlite lib32-sqlite libxcomposite lib32-libxcomposite libxinerama lib32-libgcrypt libgcrypt lib32-libxinerama ncurses lib32-ncurses opencl-icd-loader lib32-opencl-icd-loader libxslt lib32-libxslt libva lib32-libva gtk3 lib32-gtk3 gst-plugins-base-libs lib32-gst-plugins-base-libs vulkan-icd-loader lib32-vulkan-icd-loader
warning: wine-staging-7.0rc3-1 is up to date -- skipping
warning: giflib-5.2.1-2 is up to date -- skipping
warning: lib32-giflib-5.2.1-1 is up to date -- skipping
warning: libpng-1.6.37-3 is up to date -- skipping
warning: lib32-libpng-1.6.37-3 is up to date -- skipping
warning: libldap-2.6.0-2 is up to date -- skipping
warning: lib32-libldap-2.6.0-1 is up to date -- skipping
warning: gnutls-3.7.2-2 is up to date -- skipping
warning: lib32-gnutls-3.7.2-1 is up to date -- skipping
warning: mpg123-1.29.3-1 is up to date -- skipping
warning: lib32-mpg123-1.29.3-1 is up to date -- skipping
warning: openal-1.21.1-2 is up to date -- skipping
warning: lib32-openal-1.21.1-2 is up to date -- skipping
warning: v4l-utils-1.22.1-1 is up to date -- skipping
warning: lib32-v4l-utils-1.22.1-1 is up to date -- skipping
warning: libpulse-15.0-1 is up to date -- skipping
warning: lib32-libpulse-15.0-1 is up to date -- skipping
warning: libgpg-error-1.43-1 is up to date -- skipping
warning: lib32-libgpg-error-1.43-1 is up to date -- skipping
warning: alsa-plugins-1:1.2.6-1 is up to date -- skipping
warning: lib32-alsa-plugins-1.2.6-1 is up to date -- skipping
warning: alsa-lib-1.2.6.1-1 is up to date -- skipping
warning: lib32-alsa-lib-1.2.6.1-1 is up to date -- skipping
warning: libjpeg-turbo-2.1.2-1 is up to date -- skipping
warning: lib32-libjpeg-turbo-2.1.2-1 is up to date -- skipping
warning: sqlite-3.37.0-1 is up to date -- skipping
warning: lib32-sqlite-3.37.0-1 is up to date -- skipping
warning: libxcomposite-0.4.5-3 is up to date -- skipping
warning: lib32-libxcomposite-0.4.5-1 is up to date -- skipping
warning: libxinerama-1.1.4-3 is up to date -- skipping
warning: lib32-libgcrypt-1.9.4-1 is up to date -- skipping
warning: libgcrypt-1.9.4-1 is up to date -- skipping
warning: lib32-libxinerama-1.1.4-1 is up to date -- skipping
warning: ncurses-6.3-1 is up to date -- skipping
warning: lib32-ncurses-6.3-1 is up to date -- skipping
warning: ocl-icd-2.3.1-1 is up to date -- skipping
warning: lib32-ocl-icd-2.3.0-1 is up to date -- skipping
warning: libxslt-1.1.34-6 is up to date -- skipping
warning: lib32-libxslt-1.1.34-2 is up to date -- skipping
warning: libva-2.13.0-1 is up to date -- skipping
warning: lib32-libva-2.13.0-1 is up to date -- skipping
warning: gtk3-1:3.24.31-1 is up to date -- skipping
warning: lib32-gtk3-3.24.31-1 is up to date -- skipping
warning: gst-plugins-base-libs-1.18.5-1 is up to date -- skipping
warning: lib32-gst-plugins-base-libs-1.18.4-1 is up to date -- skipping
warning: vulkan-icd-loader-1.2.202-1 is up to date -- skipping
warning: lib32-vulkan-icd-loader-1.2.202-1 is up to date -- skipping
 there is nothing to do
```
### How to Reproduce

1. Select any game or software via Lutris. I've tried Terraforming mars and Blizzard Battle.net
2. Click on Install
3. Wait for game installation files to download and Lutris trying to create the wine prefix
4. Observer the dialog "updating wine prefix configuration. It will hang for a very long time.
5. After wating a very long time (several minutes) it will eventually fail with a dialog.

If I choose not to delete the wine prefix, I could install Battle.net on a second, subsequent try (the wine prefix initialized correctly). In the case of Terraformin Mars, the prefix creation never finishes.

I can also reproduce this with system wine (currently wine-staging 7.0rc3-1). I haven't extensively tried different wine versions available via Lutris.

### Expected behavior

Wine prefix creation should not fail.

### Log output

```
Started initial process 66436 from /home/ville/.local/share/lutris/runners/wine/lutris-fshack-6.21-6-x86_64/bin/wine /home/ville/.cache/Lutris-cache/terraforming-mars/gog/setup_terraforming_mars_1.3000.4.11480_(64bit)_(51903).exe /SP- /NOCANCEL
Start monitoring process.
009c:err:environ:run_wineboot boot event wait timed out
wine: could not load kernel32.dll, status c0000135
Monitored process exited.
Initial process has exited (return code: 13568)
All processes have quit
Exit with return code 13568
```
### System Information

```
[System]
OS:              Arch Linux rolling n/a
Arch:            x86_64
Kernel:          5.15.12-zen1-1-zen
Desktop:         KDE
Display Server:  x11

[CPU]
Vendor:          AuthenticAMD
Model:           AMD Ryzen 9 5950X 16-Core Processor
Physical cores:  16
Logical cores:   32

[Memory]
RAM:             31.3 GB
Swap:            0.0 GB

[Graphics]
Vendor:          AMD
OpenGL Renderer: AMD Radeon RX Vega (VEGA10, DRM 3.42.0, 5.15.12-zen1-1-zen, LLVM 13.0.0)
OpenGL Version:  4.6 (Compatibility Profile) Mesa 21.3.3
OpenGL Core:     4.6 (Core Profile) Mesa 21.3.3
OpenGL ES:       OpenGL ES 3.2 Mesa 21.3.3
Vulkan:          Supported
```

### Media (optional)

![Screenshot_20220102_151415](https://user-images.githubusercontent.com/3398736/147877293-d0f3bbfe-0257-4721-8f93-35e1eeef75a6.png)
![Screenshot_20220102_150832](https://user-images.githubusercontent.com/3398736/147877294-98a0c81d-440d-4f10-816a-79c76893d819.png)

### Checklist:

- [X] I'm not asking for support with a game or the wine runner.
- [X] I have followed the above mentioned guides and have all the graphics and wine dependencies installed.
- [X] I have checked for existing issues that describe my problem prior to opening this one.
- [X] I understand that improperly formatted bug reports may be closed without explanation.

**Comment by jrobeson** [comment]
probably not a lutris bug, but a wine bug. I've seen this same thing happen on wine occasionally, and then an update happens and the problem goes away.    But you should check if you have any wine processes left running before starting lutris, just in case. Sometimes I've had to run `pkill -9 -f winedevice` for example when I manually kill a game's process.

**Comment by WildPenquin** [comment]
Hi,

I have already done all obvious checks and workarounds, such as killing all wine processes as you've described. I do agree your tip is very sound; the very first thing I do when getting any wine trouble, is to check the process list for any wine processes (personally, I like to grep for wine).

I agree it might be a wine bug, but it is only triggered by something Lutris does (as I've described in my report). As such I wanted to point out there's little a user can do in this situation. Where can a user get more information (such as more concise logs) about what Lutris is actually doing (besides source diving?).

Personally I've found workarounds. Actually, I've found three: 1) manually create in place where Lutris would create one while installing, as this will skip the initialization by Lutris, or 2) install the game manually outside Lutris and add the game then to Lutris. These are not optimal as both will import a wine prefix created outside Lutris, which might introduce it's own problems which have nothing to do with Lutris. The 2nd way does not track the game which might already reside in a distributor-specific library, which can be a bit mess  but does retain the ability to use Lutris as a hub (3rd workaround is to not use Lutris at all).

Now personally, I don't mind that much (as I can workaround), just wanted to point out this kind of issue!

**And, in the end ... I can not reproduce this currently! So it seems whatever it was, has been gone after some upgrade on my system. I'll do a few more tests in the coming week and will report if (or not) it is really gone.**

_Thanks anyways!_

p.s. actually, the prefix is always deleted by Lutris, contrary to what I described above - that one time the prefix was left after failing seems to have been a one-off fluke. There is another issue tracked here about the forced deletion, despite the option being there and user selecting it. It seems non-functional...

**Comment by WildPenquin** [comment]
Seems like I spoke too soon, Lutris prefix creation is indeed still hanging. I've double-verified there  are no wine processes left. I'm not sure why it succeeded once?

Possibly a duplicate: #3691

**Comment by jrobeson** [comment]
i didn't see the part where you said you ran wine outside of lutris.

**Comment by WildPenquin** [comment]
Ah yes, I can see my wording was not clear on that point.

It's weird anyways, it seems that prefix creation in Lutris works on the _first run after a reboot_, but always fails after that, even for new and separate prefixes, despite no wine processes left running on the system.

I suspect something might be "leaking" from a previous prefix creation attempt environment used by Lutris or from temporary files in tmpfs, but these are just two guesses I can come up with. I'll try to figure out what is going on (which might take several days or more than a week since I'm a bit busy on the coming week); in case anyone has some ideas or pointers in the meanwhile I'd appreciate it =).

**Comment by strycore** [comment]
I had that happen to me earlier this week when I installed Minecraft Dungeons. When I tried again yesterday, everything worked fine. Really not sure what is causing this

**Comment by strycore** [comment]
Please retry with lutris 0.5.13 and lutris-GE-Proton8-7 and ask to re-open if this is still an issue.
